packages/react-instantsearch/src/core/Index.js

```diff
@@ -40,13 +40,7 @@ class Index extends Component {
      It means that with only <Index> present a new query will be sent to Algolia.
      That way you don't need a virtual hits widget to use the connectAutoComplete. 
     */
-    this.unregisterWidget = widgetsManager.registerWidget({
-      getSearchParameters: searchParameters =>
-        this.getSearchParameters(searchParameters, this.props),
-      multiIndexContext: {
```

Comment:
What this function would have return?

Reply:
The updated indexName rather than the statically computed one. But your solution works and is more obvious: when registering a widget, you pass it as a whole